### ISSUE TITLE
stdin option doesn't work for psm 0 and 2 

### ISSUE THREAD

**Issue by catethos** [tesseract-ocr/tesseract#85]
I am using the master branch from github (3.05.00dev) and facing some problems when using the stdin option in conjunction with the -psm 0 and -psm 2 options
 
```
cat <image file> | tesseract stdin stdout -psm 0
```
gives me the error

```
Error in fopenReadStream: file not found
Error in pixRead: image file not found: stdin
Cannot open input file: stdin
```

However, the command 
```
cat <image file> | tesseract stdin stdout -psm 1
```
works as expected.

Do I miss something?

Regards,
Caleb

**Comment by zdenop** [comment]
At the moment stdin is supported only for OCR.
PSM_AUTO_ONLY (0) and PSM_OSD_ONLY (2) expects to read data based on filename

**Comment by jflesch** [comment]
Just so you know, if I'm not mistaken, it worked with Tesseract 3.03. It is also very handy to script around Tesseract (avoid using temporary files).

**Comment by zdenop** [comment]
@jflesch: I am currios: Why you call tesseract executable from python instead of using C-API? IMO it should be faster...

**Comment by jflesch** [comment]
When I started working on PyOCR, it was still Tesseract 2.x. At the time, if I remember correctly, there was only a C++ API and it seemed painfully hard to use.
Also I did fork an already-existing python module.

Good to know there is a C API now. I'll have a look, see if I can bind on it easily. Thanks for the tip :)

**Comment by zdenop** [comment]
fixed with 7089c7b
